migration/src/m0000345_create_version_comparison_fns.rs

```diff
@@ -0,0 +1,61 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .get_connection()
+            .execute_unprepared(include_str!("m0000345_create_version_comparison_fns.sql"))
+            .await
+            .map(|_| ())?;
+
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .get_connection()
+            .execute_unprepared(r#"drop function semver_gt"#)
+            .await?;
+
+        manager
+            .get_connection()
+            .execute_unprepared(r#"drop function semver_gte"#)
+            .await?;
```

Comment:
At the risk of pointing out a problem to which I have no practical suggestion...

I don't love referring to the names of all the functions created in the bag of sql. It feels brittle. I get why you want to define all the functions in a single sql. I'm inclined to suggest you drop all of them in another sql, but that extra level of indirection doesn't really solve the problem of one file referring to things created in another. I'd just as soon see the names in the migration, like you have it.

Ideally, the `up()` would create all its functions in some kind of namespace and the `down()` could just obliterate the namespace (and all it contains)...

...but I dunno how to do that... so...

...nevermind.

Reply:
You're right. Imma be writing more (for other version scheme `cmp`) so lemme try to make a more-clear way (helper functions or something) to make this more directly comprehensible.